source/client-side-encryption/client-side-encryption.rst

```diff
@@ -498,13 +544,69 @@ detail described in the `Implementation`_ section:
 
       // Passed when spawning mongocryptd. If omitted, this defaults to ["--idleShutdownTimeoutSecs=60"]
       mongocryptdSpawnArgs: Optional<Array[String]>
+
+      // Set the paths used to search for the csfle library
+      csfleSearchPaths: Array<string>;
+
+      // Override the path used to load the csfle library
+      csflePathOverride: Optional<string>;
+
+      // If 'true', refuse to continue encryption without a csfle library
+      csfleRequired: boolean;
    }
 
 Drivers MUST implement extraOptions in a way that allows
 deprecating/removing options in the future without an API break, such as
 with a BSON document or map type instead of a struct type with fixed
 fields. See `Why are extraOptions and kmsProviders maps?`_.
 
+.. option:: extraOptions.csfleSearchPaths
+
+   :type: :ts:`string[]`
+   :default: :ts:`["$ORIGIN"]`
+
+   .. |opt-paths| replace:: :option:`extraOptions.csfleSearchPaths`
+
+   Allow the user to specify directories in which to search for csfle_. Refer:
+
+   - `Setting Search Paths`_
+   - `Path Resolution Behavior`_
+   - `Enabling csfle`_
+
+
+.. option:: extraOptions.csflePathOverride
+
+   :type: :ts:`undefined | string`
+   :default: :ts:`undefined`
+
+   .. |opt-path-override| replace:: :option:`extraOptions.csflePathOverride`
+
+   Allow the user to specify an absolute path to a csfle_ dynamic library to
+   load. Refer:
+
+   - `Overriding the csfle Library Path`_
+   - `Path Resolution Behavior`_
+   - `Enabling csfle`_
+
+.. option:: extraOptions.csfleRequired
+
+   :type: :ts:`boolean`
+   :default: |false|
+
+   .. |opt-csfle-required| replace:: :option:`extraOptions.csfleRequired`
+
+   If |true|, the driver MUST refuse to continue unless csfle_ was loaded
+   successfully.
+
+   If, after initializing a `libmongocrypt_handle`, csfle_ is detected to be
+   unavailable AND |opt-csfle-required| is |true|, the driver
+   MUST consider the `libmongocrypt_handle` to be invalid and return an error to
```

Comment:
Currently, the `csfle-required` flag would not be passed to the C library. The driver would instead call `mongocrypt_init()` and check for the usage of the csfle library via `mongocrypt_csfle_version_string()`, and fail if the latter reports no csfle library being loaded.

I think this is okay – the driver will have to call the two methods in that order anyway since that’s also the way to determine whether mongocryptd needs to be spawned (or at least connected to).